internal/ethapi/api.go

```diff
@@ -1003,9 +1003,12 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByHash(ctx context.Context,
 func (s *PublicTransactionPoolAPI) GetTransactionReceipt(hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index := core.GetTransaction(s.b.ChainDb(), hash)
 	if tx == nil {
-		return nil, nil
+		return nil, fmt.Errorf("Missing transaction for hash %s", hash.String())
 	}
 	receipt, _, _, _ := core.GetReceipt(s.b.ChainDb(), hash) // Old receipts don't have the lookup data available
+	if receipt == nil {
+		return nil, fmt.Errorf("Missing receipt for hash %s", hash.String())
+	}
```

Comment:
Similarly, I'd just go with `unknown receipt`.

Reply:
Although, come to think of it, shouldn't this case be impossible in the first place?

Reply:
I'd agree, but it obviously isn't, judging from the panics. We should probably look into how that can occur.

Reply:
Last time we did such a thing it turned into a very very nasty bug that's still not completely fixed (https://github.com/ethereum/go-ethereum/pull/15235).

Reply:
@karalabe
in first comment you can see mention to issues, when geth panics on getTransactionReceipt.
thats happened to me 3 times.
thats why i created this request. at least node wont shutdown on getTransactionReceipt.
i agree that something strange happened to blocks database in this case.